keel-api/src/main/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigController.kt

```diff
@@ -45,5 +55,51 @@ class DeliveryConfigController(
   fun diff(@RequestBody deliveryConfig: SubmittedDeliveryConfig): List<EnvironmentDiff> =
     adHocDiffer.calculate(deliveryConfig)
 
+  @GetMapping(
+    path = ["/{name}/environment/{environment}/constraint-states"],
+    produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
+  )
+  fun getConstraintState(
+    @PathVariable("name") name: String,
+    @PathVariable("environment") environment: String,
+    @RequestParam("limit") limit: Int?
+  ): List<ConstraintState> =
+    deliveryConfigRepository.constraintStateFor(name, environment, limit ?: DEFAULT_MAX_EVENTS)
+
+  @PostMapping(
+    path = ["/{name}/environment/{environment}/constraint-state/{type}"],
+    consumes = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE],
+    produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
+  )
+  // TODO: Keel needs fiat integration and should use kork's AuthenticatedRequest class.
```

Comment:
This integration already exists and is used in the `ResourceController`: ([here](https://github.com/spinnaker/keel/blob/master/keel-api/src/main/kotlin/com/netflix/spinnaker/keel/rest/ResourceController.kt#L79) is an example of it in action)

Reply:
Unless you're talking about something else? I agree that service account probably needs to be able to be pulled to a higher level

Reply:
I missed that, thanks! I'm going to leave a TODO here until we move service account to a higher level, but will update its text.